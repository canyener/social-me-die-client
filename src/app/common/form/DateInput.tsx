import { FieldRenderProps } from 'react-final-form'
import { Form, FormFieldProps, Label } from 'semantic-ui-react'
import { DateTimePicker } from 'react-widgets'

interface IProps
  extends FieldRenderProps<Date, HTMLElement>,
  FormFieldProps {
  id?: string
}

const DateInput: React.FC<IProps> = ({
  input,
  width,
  placeholder,
  date = false,
  time = false,
  meta: { touched, error },
  ...rest
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <DateTimePicker
        placeholder={placeholder}
        value={input.value || null}
        onChange={input.onChange}
        date={date}
        time={time}
        {...rest}
      />
      {
        touched && error && (
          <Label basic color='red'>{error}</Label>
        )
      }
    </Form.Field>
  )
}

export default DateInput