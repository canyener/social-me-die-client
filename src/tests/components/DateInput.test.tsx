import { shallow } from 'enzyme'
import DateInput from "../../app/common/form/DateInput"

test('Should render DateInput correctly without any given errors', () => {

  const wrapper = shallow(<DateInput
    id='test-id'
    width='11'
    placeholder='Test Input PlaceHolder'
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test input name',
      value: 'Test input value'
    }}
    meta={{
      touched: false,
      error: undefined
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})

test('Should set Form.Field error to true and render Label with given error message', () => {
  const wrapper = shallow(<DateInput
    type='text'
    width='11'
    placeholder='Test Input PlaceHolder'
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test input name',
      value: 'Test input value'
    }}
    meta={{
      touched: true,
      error: 'test error message'
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})