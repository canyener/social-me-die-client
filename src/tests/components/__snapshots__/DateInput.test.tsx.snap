// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DateInput correctly without any given errors 1`] = `
<FormField
  error={false}
  width="11"
>
  <Uncontrolled(DateTimePicker)
    id="test-id"
    onChange={[MockFunction]}
    placeholder="Test Input PlaceHolder"
    value="Test input value"
  />
</FormField>
`;

exports[`Should set Form.Field error to true and render Label with given error message 1`] = `
<FormField
  error={true}
  width="11"
>
  <Uncontrolled(DateTimePicker)
    onChange={[MockFunction]}
    placeholder="Test Input PlaceHolder"
    type="text"
    value="Test input value"
  />
  <Label
    basic={true}
    color="red"
  >
    test error message
  </Label>
</FormField>
`;
